programs/editor/Tools.js

```diff
@@ -62,24 +62,32 @@ define("webodf/editor/Tools", [
                 aboutDialog,
                 sessionSubscribers = [];
 
+            function placeAndStartUpWidget(widget) {
+                widget.placeAt(toolbar);
+                widget.startup();
+            }
+
             /**
              * Creates a tool and installs it, if the enabled flag is set to true.
              * Only supports tool classes whose constructor has a single argument which
              * is a callback to pass the created widget object to.
              * @param {!function(new:Object, function(!Object):undefined)} Tool  constructor method of the tool
              * @param {!boolean} enabled
+             * @param {!Object|undefined=} config
              * @return {?Object}
              */
-            function createTool(Tool, enabled) {
+            function createTool(Tool, enabled, config) {
                 var tool = null;
 
                 if (enabled) {
-                    tool = new Tool(function (widget) {
-                        widget.placeAt(toolbar);
-                        widget.startup();
-                    });
+                    if (config) {
+                        tool = new Tool(config, placeAndStartUpWidget);
```

Comment:
Hm, `placeAndStartUpWidget` is a callback to be called when the tool is done with init/setup. So far it seemed to me `config, cbDone` would be a quite usual order of parameters.
I agree that the code needed to support both configless and config-using constructors is not nice here, but I would prefer it over having a non-usual order of constructor parameters
(though writing that I remember I recently heard callback parameters to constructors are bad per se, but forgot the reason, and with Dojo we have that anyway).